include/zephyr/devicetree.h

```diff
@@ -551,6 +551,14 @@
  */
 #define DT_NODE_CHILD_IDX(node_id) DT_CAT(node_id, _CHILD_IDX)
 
+/**
+ * @brief Get the number of child nodes of a given node
+ *
+ * @param node_id a node identifier
+ * @return Number of child nodes
+ */
+#define DT_CHILD_NUM(node_id) DT_CAT(node_id, _CHILD_NUM)
```

Comment:
Perhaps we should align with the `FOREACH_CHILD` macros, and add `DT_CHILD_NUM_STATUS_OKAY` too.

Reply:
Sure, it has been added.